rasterio/rio/info.py

```diff
@@ -15,29 +15,153 @@
 from rasterio.transform import guard_transform
 
 
+# Handlers for info module options.
+
+def from_like_context(ctx, param, value):
+    """Return the value for an option from the context if the option 
+    or `--all` is given, else return None."""
+    if ctx.obj and ctx.obj.get('like') and (
+            value == 'like' or ctx.obj.get('all_like')):
+        return ctx.obj['like'][param.name]
+    else:
+        return None
+
+
+def all_handler(ctx, param, value):
+    """Get tags from a template file or command line."""
+    if ctx.obj and ctx.obj.get('like'):
+        ctx.obj['all_like'] = value
+        value = ctx.obj.get('like')
+    return value
+
+
+def crs_handler(ctx, param, value):
+    """Get crs value from a template file or command line."""
+    retval = from_like_context(ctx, param, value)
+    if retval is None and value:
+        try:
+            retval = json.loads(value)
+        except ValueError:
+            retval = value
+        if not (rasterio.crs.is_geographic_crs(retval) or
```

Comment:
@sgillies consider adding a ```rasterio.crs.is_valid_crs``` function to handle this objective (internally it can make these two calls).  It seems a little out of place to use these tests because in this context we don't care if it is geographic or projected, we only care that it is valid...